indigo_app/views/works.py

```diff
@@ -55,32 +55,48 @@ def get_context_data(self, **kwargs):
         context['work_json'] = json.dumps(WorkSerializer(instance=self.work, context={'request': self.request}).data)
         return context
 
-    def get_work_timeline(self):
-        # add other dates to timeline
-        work_timeline = self.work.points_in_time()
+    def get_work_timeline(self, work):
+        # get initial, amendment, consolidation dates
+        dates = work.possible_expression_dates()
+        # add assent, publication, repeal, commencement dates
         other_dates = [
-            ('assent_date', self.work.assent_date),
-            ('publication_date', self.work.publication_date),
-            ('repealed_date', self.work.repealed_date)
+            ('assent_date', work.assent_date),
+            ('publication_date', work.publication_date),
+            ('repealed_date', work.repealed_date)
         ]
+        for c in work.commencements.all():
+            other_dates.append(('commencement_date', c.date))
 
-        # add new events (e.g. if assent is before any of the other events),
-        # only one event per date
         for name, date in other_dates:
             if date:
-                dates = [entry['date'] for entry in work_timeline]
-                if date not in dates:
-                    work_timeline.append({
+                if date not in [entry['date'] for entry in dates]:
+                    dates.append({
                         'date': date,
                         name: True,
                     })
                 else:
                     # add to existing events (e.g. if publication and commencement dates are the same)
-                    for entry in work_timeline:
+                    for entry in dates:
                         if entry['date'] == date:
                             entry[name] = True
 
-        return sorted(work_timeline, key=lambda k: k['date'], reverse=True)
+        dates.sort(key=lambda x: x['date'], reverse=True)
+
+        # add expressions and commencement and amendment objects
+        work_timeline = [{
```

Comment:
Rather than copying everything from the other dict, would it be easier to:

```python
for event in dates:
    event.commencements = ...
    etc.
```

Reply:
um yes that's much better thank you!